File: baekjoon_all/10000+/boj_10866_solution1.cpp
// Solve 2022-06-06
// Update 2023-06-26

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

int main() {
    FASTIO;

    int n;
    cin >> n;

    deque<int> dq;

    for (int i = 0; i < n; i++) {
        string cmd;
        cin >> cmd;

        if (cmd == "push_front") {
            int x;
            cin >> x;
            dq.push_front(x);
        }
        else if (cmd == "push_back") {
            int x;
            cin >> x;
            dq.push_back(x);
        }
        else if (cmd == "pop_front") {
            if (dq.empty()) {
                cout << -1 << '\n';
            }
            else {
                cout << dq.front() << '\n';
                dq.pop_front();
            }
        }
        else if (cmd == "pop_back") {
            if (dq.empty()) {
                cout << -1 << '\n';
            }
            else {
                cout << dq.back() << '\n';
                dq.pop_back();
            }
        }
        else if (cmd == "size") {
            cout << SIZE(dq) << '\n';
        }
        else if (cmd == "empty") {
            cout << dq.empty() << '\n';
        }
        else if (cmd == "front") {
            if (dq.empty()) {
                cout << -1 << '\n';
            }
            else {
                cout << dq.front() << '\n';
            }
        }
        else {
            if (dq.empty()) {
                cout << -1 << '\n';
            }
            else {
                cout << dq.back() << '\n';
            }
        }
    }

    return 0;
}

File: baekjoon_all/10000+/boj_10866_solution1.py
# Solve 2023-03-01

import sys
from collections import deque

input = lambda : sys.stdin.readline().rstrip()

n = int(input())
dq = deque()

for _ in range(n):
    line = input().split()

    if line[0] == "push_front":
        dq.appendleft(line[1])
    elif line[0] == "push_back":
        dq.append(line[1])
    elif line[0] == "pop_front":
        if len(dq) == 0:
            print(-1)
        else:
            print(dq.popleft())
    elif line[0] == "pop_back":
        if len(dq) == 0:
            print(-1)
        else:
            print(dq.pop())
    elif line[0] == "size":
        print(len(dq))
    elif line[0] == "empty":
        if len(dq) == 0:
            print(1)
        else:
            print(0)
    elif line[0] == "front":
        if len(dq) == 0:
            print(-1)
        else:
            print(dq[0])
    else:
        if len(dq) == 0:
            print(-1)
        else:
            print(dq[-1])
